feat(jetbrains): forward extension host logs to JetBrains

hassoncs opened the pull request:
Introduces a new utility (`fowardingLogger.ts`) that monkey-patches the global `console` methods to forward messages to the JetBrains backend via RPC. This forwarding is conditionally registered in `extension.ts` when the extension is running in a Kilo Code wrapped environment (i.e., JetBrains).

The JetBrains plugin (`MainThreadConsoleShape.kt`) is updated to correctly receive and process these logs, including specific handling for error severity. A noisy log in `RPCProtocol.kt` was also commented out.

Commit:
feat(jetbrains): forward extension host logs to JetBrains

Introduces a new utility (`fowardingLogger.ts`) that monkey-patches the global `console` methods to forward messages to the JetBrains backend via RPC. This forwarding is conditionally registered in `extension.ts` when the extension is running in a Kilo Code wrapped environment (i.e., JetBrains).

The JetBrains plugin (`MainThreadConsoleShape.kt`) is updated to correctly receive and process these logs, including specific handling for error severity. A noisy log in `RPCProtocol.kt` was also commented out.

Commit:
fix: resolve conflicting Git configuration for platform.zip

Problem:
- platform.zip was simultaneously tracked in Git LFS (.gitattributes)
  AND ignored (.gitignore), creating contradictory instructions
- This caused unpredictable behavior where the file would appear/disappear
  during development builds
- The file is a 100+ MB generated build artifact containing merged VSCode
  platform node_modules for multiple architectures

Resolution:
- Removed Git LFS tracking from .gitattributes for jetbrains/plugin/platform.zip
- Kept the file ignored in jetbrains/plugin/.gitignore
- Now treated as pure build artifact generated via `./gradlew genPlatform`
- Developers only need to generate it for release builds (debugMode=release)
- Development builds (debugMode=idea/none) don't require this file

The file can be regenerated anytime using the genPlatform gradle task,
eliminating the need for version control tracking and reducing repository size.

catrielmuller commented on jetbrains/plugin/platform.zip:
Please don't delete this file

catrielmuller commented on .gitattributes:
```diff
@@ -17,5 +17,3 @@ webview-ui/src/i18n/locales/** linguist-generated=true
 src/i18n/locales/en/** linguist-generated=false
 webview-ui/src/i18n/locales/en/** linguist-generated=false
-# This approach uses gitattributes' last-match-wins rule to exclude English while including all other locales
-jetbrains/plugin/platform.zip filter=lfs diff=lfs merge=lfs -text
```
this is required for the final bundle

catrielmuller reviewed (changes requested):
LGTM, please restore the platform.zip

Commit:
Revert "fix: resolve conflicting Git configuration for platform.zip"

This reverts commit a39eabaad74b0b15f0ca75316e47a62f6c5639a8.

Commit:
Remove platform.zip from .gitignore since it's checked in

Commit:
build(jetbrains): ensure platform.zip is deleted before generation

The deletion of `platform.zip` has been moved from the end of the `download` function to the `doFirst` block of the `genPlatform` task. This ensures that the zip file is deleted prior to the download and generation process, preventing potential issues with stale files.
